### ISSUE TITLE
LLVM ERROR: Broken module found, compilation aborted! Instruction does not dominate all uses.

### ISSUE THREAD

**Issue by danilaml** [llvm/llvm-project#69196]
After https://github.com/llvm/llvm-project/commit/ea7f43ec14c5522cec7b787c5fc3c323f7459e89
the following IR:
```llvm
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128-ni:1-p2:32:8:8:32-ni:2"
target triple = "x86_64-unknown-linux-gnu"

define void @foo() {
bb:
  br label %bb1

bb1:                                              ; preds = %bb4, %bb
  %phi = phi i32 [ 0, %bb ], [ %or, %bb4 ]
  %phi2 = phi i32 [ 0, %bb ], [ %or3, %bb4 ]
  %and = and i32 0, 0
  %shl = shl i32 %phi, %and
  %or = or i32 %shl, 0
  %call = call i64 null(i32 %or)
  %or3 = or i32 %phi2, 0
  br label %bb4

bb4:                                              ; preds = %bb1
  br i1 false, label %bb5, label %bb1

bb5:                                              ; preds = %bb4
  %phi6 = phi i32 [ %shl, %bb4 ]
  %phi7 = phi i32 [ %or3, %bb4 ]
  ret void
}
```
crashes with assertion above when run with `opt  -S -passes=slp-vectorizer -slp-threshold=-9999`.
Godbolt: https://godbolt.org/z/KW4bMx4qM

**Comment by alexey-bataev** [comment]
Fixed in 8e2b2c4181506efc5b9321c203dd107bbd63392b

**Comment by alexey-bataev** [comment]
Need to fix other crashes, reopenning

**Comment by alexey-bataev** [comment]
Fixed in 66775f8ccdcc8264ef349518e1c59d96d4227823
